commit fe40ba8aa6e61beb9e3bf89742df2b66091739ea
Author: David Jarvie <djarvie@kde.org>
Date:   Thu May 23 18:51:33 2019 +0100

    Bug 407544: Fix crash when creating new KAlarm resource
    
    The alarm type specification in the resource edit dialog was handled
    incorrectly since refactoring, and caused a crash.

diff --git a/resources/kalarm/kalarm/kalarmconfig.cpp b/resources/kalarm/kalarm/kalarmconfig.cpp
index 2cb20d34c..b7ee23ea4 100644
--- a/resources/kalarm/kalarm/kalarmconfig.cpp
+++ b/resources/kalarm/kalarm/kalarmconfig.cpp
@@ -1,5 +1,6 @@
 /*
  *  Copyright © 2018  Daniel Vrátil <dvratil@kde.org>
+ *  Copyright © 2019  David Jarvie <djarvie@kde.org>
  *
  *  This library is free software; you can redistribute it and/or modify it
  *  under the terms of the GNU Library General Public License as published by
@@ -38,6 +39,12 @@ public:
         mWidget->setMonitorEnabled(false);
     }
 
+    bool save() const override
+    {
+        mSettings->setAlarmTypes(CalEvent::mimeTypes(mTypeSelector->alarmType()));
+        return SingleFileResourceConfigBase<SETTINGS_NAMESPACE::Settings>::save();
+    }
+
 private:
     QScopedPointer<AlarmTypeRadioWidget> mTypeSelector;
 };
diff --git a/resources/kalarm/kalarm/kalarmresource.cpp b/resources/kalarm/kalarm/kalarmresource.cpp
index cd763e9c7..9d7c4969b 100644
--- a/resources/kalarm/kalarm/kalarmresource.cpp
+++ b/resources/kalarm/kalarm/kalarmresource.cpp
@@ -1,7 +1,7 @@
 /*
  *  kalarmresource.cpp  -  Akonadi resource for KAlarm
  *  Program:  kalarm
- *  Copyright © 2009-2014 by David Jarvie <djarvie@kde.org>
+ *  Copyright © 2009-2019 by David Jarvie <djarvie@kde.org>
  *
  *  This library is free software; you can redistribute it and/or modify it
  *  under the terms of the GNU Library General Public License as published by
@@ -65,15 +65,6 @@ KAlarmResource::~KAlarmResource()
 {
 }
 
-/******************************************************************************
-* Save extra settings after the configuration dialog has been accepted.
-*/
-void KAlarmResource::applyConfigurationChanges()
-{
-    mSettings->setAlarmTypes(CalEvent::mimeTypes(mTypeSelector->alarmType()));
-    mSettings->save();
-}
-
 /******************************************************************************
 * Reimplemented to fetch collection attributes after creating the collection.
 */
diff --git a/resources/kalarm/kalarm/kalarmresource.h b/resources/kalarm/kalarm/kalarmresource.h
index 825b644d5..3020ca250 100644
--- a/resources/kalarm/kalarm/kalarmresource.h
+++ b/resources/kalarm/kalarm/kalarmresource.h
@@ -1,7 +1,7 @@
 /*
  *  kalarmresource.h  -  Akonadi resource for KAlarm
  *  Program:  kalarm
- *  Copyright © 2009-2014 by David Jarvie <djarvie@kde.org>
+ *  Copyright © 2009-2019 by David Jarvie <djarvie@kde.org>
  *
  *  This library is free software; you can redistribute it and/or modify it
  *  under the terms of the GNU Library General Public License as published by
@@ -57,20 +57,17 @@ private Q_SLOTS:
     void updateFormat(KJob *);
     void setCompatibility(KJob *);
 
-protected:
-    void applyConfigurationChanges() override;
-
 private:
     void checkFileCompatibility(const Akonadi::Collection & = Akonadi::Collection(), bool createAttribute = false);
     Akonadi::CollectionFetchJob *fetchCollection(const char *slot);
 
     AlarmTypeRadioWidget *mTypeSelector = nullptr;
     KACalendar::Compat mCompatibility;
-    KACalendar::Compat mFileCompatibility;     // calendar file compatibility found by readFromFile()
-    int mVersion;                              // calendar format version
-    int mFileVersion;                          // calendar format version found by readFromFile()
-    bool mHaveReadFile;                        // the calendar file has been read
-    bool mFetchedAttributes;                   // attributes have been fetched after initialisation
+    KACalendar::Compat mFileCompatibility;  // calendar file compatibility found by readFromFile()
+    int mVersion;                           // calendar format version
+    int mFileVersion;                       // calendar format version found by readFromFile()
+    bool mHaveReadFile;                     // the calendar file has been read
+    bool mFetchedAttributes;                // attributes have been fetched after initialisation
 };
 
 #endif
